src/main/java/com/redhat/devtools/lsp4ij/server/definition/launching/UserDefinedClientFeatures.java

```diff
@@ -32,8 +33,15 @@ public UserDefinedClientFeatures() {
     }
 
     public boolean isCaseSensitive(@NotNull PsiFile file) {
-        UserDefinedLanguageServerDefinition serverDefinition = (UserDefinedLanguageServerDefinition) getServerDefinition();
+        ClientConfigurableLanguageServerDefinition serverDefinition = (ClientConfigurableLanguageServerDefinition) getServerDefinition();
         ClientConfigurationSettings clientConfiguration = serverDefinition.getLanguageServerClientConfiguration();
-        return (clientConfiguration != null) && clientConfiguration.caseSensitive;
+        return clientConfiguration != null ? clientConfiguration.caseSensitive : super.isCaseSensitive(file);
+    }
+
+    @Override
+    public boolean serverReportsOffsetOnlyPositions(@NotNull PsiFile file) {
```

Comment:
And in client-configurable language server definitions, we try to derive the value for this setting from client config.